src/WWT.Providers/Providers/tilesProvider.cs

```diff
@@ -15,35 +22,33 @@ public override void Run(IWwtContext context)
             int tileX = Convert.ToInt32(values[1]);
             int tileY = Convert.ToInt32(values[2]);
             string file = values[3];
-            string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
+            string wwtTilesDir = _options.WwtTilesDir;
 
             context.Response.AddHeader("Cache-Control", "public, max-age=31536000");
             context.Response.AddHeader("Expires", "Thu, 31 Dec 2009 16:00:00 GMT");
             context.Response.AddHeader("ETag", "155");
             context.Response.AddHeader("Last-Modified", "Tue, 20 May 2008 22:32:37 GMT");
-            //context.Response.AddHeader("Cache-Control", "max-age=36000");
-
 
             if (level < 8)
             {
                 context.Response.ContentType = "image/png";
-                Stream s = PlateTilePyramid.GetFileStream(String.Format(wwtTilesDir + "\\{0}.plate", file), level, tileX, tileY);
-                int length = (int)s.Length;
-                if (length == 0)
-                {
 
-                    context.Response.Clear();
-                    context.Response.ContentType = "text/plain";
-                    context.Response.Write("No image");
-                    context.Response.End();
-                    return;
+                using (Stream s =_plateTiles.GetStream(wwtTilesDir, $"{file}.plate", level, tileX, tileY))
```

Comment:
Note that this endpoint generates its filename from an arbitrary input, so we should think about security implications here. Given our particular case, though, I can't think of anything especially worrisome.

Reply:
Good point. I had thought about it and written that down as a follow up.